include/igl/predicates/ear_clipping.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include <igl/slice.h>
+#include "ear_clipping.h"
+#include "point_inside_polygon.h"
+#include "predicates.h"
+
+template <typename DerivedP>
+IGL_INLINE bool igl::predicates::is_ear(
+  const Eigen::MatrixBase<DerivedP>& P,
+  const Eigen::VectorXi& RT,
```

Comment:
template with MatrixBase<DerivedRT>
```suggestion
  const Eigen::MatrixBase<DerivedRT>& RT,
```